Abbyy.CloudSdk.V2.Client.Sample/Program.cs

```diff
@@ -32,35 +37,105 @@ public class Program
 		/// </summary>
 		private const string ServiceUrl = "https://cloud-eu.ocrsdk.com";
 
-		private static AuthInfo _authInfo;
+		private static int _retryCount = 3;
+		private static int _delayBetweenRetriesInSeconds = 3;
+		private static string _httpClientName = "OCR_HTTP_CLIENT";
+
+		private static readonly AuthInfo AuthInfo = new AuthInfo
+		{
+			Host = ServiceUrl,
+			ApplicationId = ApplicationId,
+			Password = Password
+		};
+
+		private static ServiceProvider _serviceProvider;
+		private static HttpClient _httpClient;
 
 		public static async Task Main()
 		{
-			// Init library
-			_authInfo = new AuthInfo
+			// Init client
+			// You could also call GetOcrClient to use client without retry policy
+			using (var ocrClient = GetOcrClientWithRetryPolicy())
 			{
-				Host = ServiceUrl,
-				ApplicationId = ApplicationId,
-				Password = Password,
-			};
-			var ocrClient = new OcrClient(_authInfo);
+				// Process image
+				// You could also call ProcessDocumentAsync or any other processing method declared below
+				var resultUrls = await ProcessImageAsync(ocrClient);
 
-			// Process image
-			// You could also call ProcessDocumentAsync or any other processing method declared below
-			var resultUrls = await ProcessImageAsync(ocrClient);
+				// Get results
+				foreach (var resultUrl in resultUrls)
+					Console.WriteLine(resultUrl);
 
-			// Get results
-			foreach (var resultUrl in resultUrls)
-			{
-				Console.WriteLine(resultUrl);
+				var finishedTasks = await GetFinishedTasksWithRetry();
```

Comment:
Inside first using (var ocrClient ...) block in the method GetFinishedTasksWithRetry the second block  using(var ocrClient...) is used. Maybe transfer the second method beyond the first block of using?

Reply:
Yes, I think we should delete this code and replace GetFinishedTasksWithRetry with GetFinishedTasks method, the client will be an argument, just like in ProcessImage method

Reply:
Fixed